### DISCUSSION TITLE
Scheduler is missing after running for a while

### DISCUSSION THREAD

**Discussion by RPJack** [quartznet/quartznet#1504] [Q&A]
Hello everyone, i am using Qratznet on my product web application first at Dec 03,2021. i’m running my job that send file to SFTP server ervery day 06:00 AM, and it did a great job.
But the job was stopped at Feb 01,2022. I was checked the logs, the schduler was stopped at Jan 31th when the current job was done, there is screenshot for JobRunnerShell Logs：
![image](https://user-images.githubusercontent.com/31195817/152158955-74dd2dc6-1c53-42a9-b7e2-9726bfdf2557.png)

I used one schduler and multiple jobs on asp net core 2.1, but the job cronexpression is different like: [0 0 6 * * ?] or [5 0 6 * * ?], and the job running concurrency is 3.
I don't konw why the JobRunnerShell and Scheduler's log is missing, it maybe the key that my job was stoped at next day. Could you give me some help and advice?

**Comment by lahma** [comment]
Does your site have always on configured so that it won't go to sleep? Do your triggers have end date set? There's no reason why Quartz would stop running by itself.

**Reply by RPJack** [reply]
No, it's the website that has configured view and scheduler is running background forever until the website was stopped. And i published so much container use the same code, it's the only one that schduler stopped by itself. There is the different for others that it have many jobs at the period of time. Is there has problem that when i used one trigger configured many jobs?

**Reply by lahma** [reply]
Well in a way it's odd, one trigger can refer to single job after all.

**Reply by RPJack** [reply]
Sorry, it's my mistake, I mean that one scheduler configured many jobs

**Reply by RPJack** [reply]
Maybe it also relates the environment, i can't reproduce it on my development environment

**Reply by lahma** [reply]
Which will make it even more guesswork I'm afraid.

**Reply by RPJack** [reply]
Yeah, it's hard to find the question key know, i will log as much detail as possible if the similar question happen again.

**Reply by RPJack** [reply]
Thank you for your reply!

**Reply by sabarirajang** [reply]
We do face similar problem. We have 5 jobs added to a scheduler to run at different intervals. 1 to run at every minute, 1 at 12 PM and 3 of them at 5 PM. The job which runs at every minute is running without issues  but the other 4 jobs stopped.

**Reply by lahma** [reply]
I these cases logging can be quite helpful.

**Reply by RPJack** [reply]
@sabarirajang Is your Job Scheduler stopped after the last job running? you can open debug log to analyze the logger who Quartz.Core.JobRunShell and Quartz.Core.QuartzSchedulerThread, i think it's helpful to fix it

**Reply by sabarirajang** [reply]
Thank you @lahma & @RPJack for your responses.
@RPJack I think the job scheduler is not stopped because one of the 5 jobs is still running on that scheduler. The other 4 jobs stopped after the last run. Unfortunately our current implementation did not enable logging from Quartz. I will try to add logging and see if this issue occurs again.
